Libs/MRML/Core/vtkMRMLScene.cxx

```diff
@@ -1350,6 +1350,7 @@ void vtkMRMLScene::RemoveNode(vtkMRMLNode *n)
     this->RemoveReferencesToNode(n);
     }
 
+  n->SetID(NULL);
```

Comment:
That's what I wanted to do, but SetID is a private method. Any suggestions?

My line of thinking was that if a node is removed from the scene, then its ID - which is scene-specific - will be unused, and then re-assigned anyway when it is added to another scene. And without a scene, the ID is unused. So I think there should be no side-effects, but I cannot be 100% sure...